packages/server/src/routes/grantsSavedSearch.js

```diff
@@ -0,0 +1,63 @@
+const express = require('express');
+
+const router = express.Router({ mergeParams: true });
+const db = require('../db');
+const { requireUser } = require('../lib/access-helpers');
+
+router.get('/', requireUser, async (req, res) => {
+    const { user } = req.session;
+
+    const paginationParams = {
+        currentPage: req.params.currentPage || 1,
+        perPage: req.params.perPage || 10,
+        isLengthAware: req.params.isLengthAware || true,
+    };
+
+    const savedSearches = await db.getSavedSearches(user.id, paginationParams);
+
+    res.json(savedSearches);
+});
+
+router.post('/', requireUser, async (req, res) => {
+    const { user } = req.session;
+
+    try {
+        const result = await db.createSavedSearch({
+            name: req.body.name,
+            userId: user.id,
+            criteria: req.body.criteria,
+        });
+
+        res.json(result);
+    } catch (e) {
+        res.status(500).send('Unable to create saved search. Please reach out to grants-helpdesk@usdigitalresponse.org');
+    }
+});
+
+router.delete('/:searchId', requireUser, async (req, res) => {
+    const { searchId } = req.params;
+    const { user } = req.session;
+
+    const toDelete = await db.getSavedSearch(searchId);
+    if (!toDelete || toDelete.created_by !== user.id) {
+        res.sendStatus(400).send('Could not find the saved search');
+        return;
+    }
+    let deleteSuccess = false;
+
+    try {
+        deleteSuccess = await db.deleteSavedSearch(toDelete.id);
+    } catch (e) {
+        console.error(`Error deleting saved search: ${e}`);
+        res.status(500).send('Error deleting saved search');
+        return;
+    }
```

Comment:
This could be simplified a bit with a small tweak to the `db.deleteSavedSearch` function, which would have the benefits of atomicity, avoiding the `400` response case, and not confirming/denying the validity of a saved search ID to anyone other than the user who created it (in other words, it can only be "found" by its creator).

In `packages/server/src/db/index.js`:
```js
async function deleteSavedSearch(searchId, createdBy=null) {
    let rowsDeleted = 0;

    const params = { id: searchId }
    if createdBy !== null {
        params["created_by"] = createdBy;
    }


    try {
        rowsDeleted = await knex('grants_saved_searches')
            .where(params)
            .del();
    } catch (e) {
        console.error(`Unable to delete ${searchId}. Error: ${e}`);
        throw e;
    }

    return rowsDeleted === 1;
}
```

In `packages/server/src/routes/grantsSavedSearch.js`:
```js
router.delete('/:searchId', requireUser, async (req, res) => {
    const { searchId } = req.params;
    const { user } = req.session;

    let deleteSuccess = false;

    try {
        deleteSuccess = await db.deleteSavedSearch(searchId, user.id);
    } catch (e) {
        console.error(`Error deleting saved search: ${e}`);
        res.status(500).send('Error deleting saved search');
        return;
    }

    if (deleteSuccess) {
        res.status(200).send('OK');
    } else {
        res.status(404).send('Could not find the saved search');
    }
});
```

Reply:
@TylerHendrickson since the current use-case for delete is only by a user. I'm going to make the `createdBy` field mandatory rather than optional. It also simplifies the tests for now.